lib/pbench/test/unit/server/test_indexer.py

```diff
@@ -52,3 +55,27 @@ def test_value_types():
             templ, {"str": "abc", "int": 123, "float": 45.6789012, "other": []}
         )
         assert res == "abc_123_45.678901_%other%_UID"
+
+
+def test_init_indexing(monkeypatch, server_config, make_logger):
+    called = [False]
+
+    class MockPbenchTemplates:
+        def update_templates(self, *args, **kwargs):
+            called[0] = True
+            assert args[0] == "fake-es-obj", f"args={args!r}, kwargs={kwargs!r}"
+            assert not kwargs
+
+    class MockIdxContext:
+        def __init__(self, *args, **kwargs):
+            self.args = args
+            self.kwargs = kwargs
+            self.templates = MockPbenchTemplates()
+            self.es = "fake-es-obj"
+
+    monkeypatch.setattr(pbench.server.indexer, "IdxContext", MockIdxContext)
+    try:
+        init_indexing("test", server_config, make_logger)
+    except Exception as exc:
+        pytest.fail(f"Unexpected exception raised: {exc}")
```

Comment:
Do you really want this `try` block?  If you omit it, won't the test fail automatically if `init_indexing()` raises an exception?

Reply:
It will, but it will be an error.  Here I am trying to communicate explicitly that we should have constructed a scenario where `init_indexing()` does not raise an exception.

Reply:
> it will be an error

Is "error" different from "fail"?

> I am trying to communicate explicitly

I'm all in favor of explicit communication, but if the test fails/errors, then I think it's clear that an unexpected exception was raised, and it's not obvious to me how calling `pytest.fail()` improves the situation.  But, it's only a few extra lines.